packages/ui5-proxy-middleware/src/ui5/middleware.ts

```diff
@@ -0,0 +1,45 @@
+import express, { RequestHandler } from 'express';
+import { MiddlewareParameters, ProxyConfig } from '../base/types';
+import { ui5Proxy } from '../base/proxy';
+import { getCorporateProxyServer, isHostExcludedFromProxy } from '../base/utils';
+import { Options } from 'http-proxy-middleware';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+import { ToolsLogger } from '@sap-ux/logger';
+import { UI5ToolingTransport } from '@sap-ux/logger/dist/transports';
+
+module.exports = ({ options }: MiddlewareParameters<ProxyConfig>): RequestHandler => {
+    const logger = new ToolsLogger({
+        transports: [new UI5ToolingTransport({ moduleName: 'ui5-proxy-middleware' })]
+    });
+    const router = express.Router();
+    const config = options.configuration;
+    const secure = !!config.secure;
+    const debug = !!config.debug;
+    const noProxyVal = process.env.no_proxy || process.env.npm_config_noproxy;
+    const corporateProxyServer = getCorporateProxyServer(config.proxy);
+    const proxyInfo = corporateProxyServer
```

Comment:
Could you please add a comment here explaining what we replace?

Reply:
Done